From a32b7559eb007f4e361fae63517e3e5c2ee1ef72 Mon Sep 17 00:00:00 2001
From: Scovotto Davide <scovottodavide@gmail.com>
Date: Wed, 20 Apr 2022 17:22:17 +0200
Subject: [PATCH] tpm_tis_spi_main: Force probe routine to run
 synchronously with driver and device registration when IMA is enabled

Co-authored-by: Alberto Solavagione <albertosolavagione30@gmail.com>
Co-developed-by: Alberto Solavagione <albertosolavagione30@gmail.com>
Signed-off-by: Alberto Solavagione <albertosolavagione30@gmail.com>
Signed-off-by: Davide Scovotto <scovottodavide@gmail.com>
---
 drivers/char/tpm/tpm_tis_spi_main.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/char/tpm/tpm_tis_spi_main.c
+++ b/drivers/char/tpm/tpm_tis_spi_main.c
@@ -294,7 +294,11 @@ static struct spi_driver tpm_tis_spi_dri
 		.pm = &tpm_tis_pm,
 		.of_match_table = of_match_ptr(of_tis_spi_match),
 		.acpi_match_table = ACPI_PTR(acpi_tis_spi_match),
+#ifdef CONFIG_IMA
+		.probe_type = PROBE_FORCE_SYNCHRONOUS,
+#else
 		.probe_type = PROBE_PREFER_ASYNCHRONOUS,
+#endif
 	},
 	.probe = tpm_tis_spi_driver_probe,
 	.remove = tpm_tis_spi_remove,
